### PR TITLE
Stats Date Picker: Design Tweaks

### PR DESCRIPTION
**Pull request by annacmc**
Related to #83269

## Proposed Changes

* updating cancel and apply button spacing
* adjusting shortcut tick position by extending the width and adding more white space
* adjusting border-radius and letter spacing to match design requirements
* updating HTML order or fix tab order
* adding aria labels
* adding 2px between shortcuts to avoid clipping of an outline when hovering the next item

## Testing Instructions

* open the traffic page on the live branch
* make sure that date picker works and looks the same as on the screenshot

SCR-20231024-mnsj

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/components/stats-date-control/stats-date-control-picker.tsx [hunk 1]
```diff
@@ -77,8 +77,14 @@ const DateControlPicker = ( {
 				position="bottom"
 				context={ infoReferenceElement?.current }
 				isVisible={ popoverOpened }
+				className="stats-date-control-picker__popover-wrapper"
 			>
 				<div className="stats-date-control-picker__popover-content">
+					<DateControlPickerShortcuts
```
Comment (kangzj) [inline client/components/stats-date-control/stats-date-control-picker.tsx:83]: Can I know more about why we are changing the order here and set direction to `row-reverse`? Thanks

**Review by kangzj — APPROVED** [review]
It works as described. Left a question, no blockers. 👍

File: client/components/stats-date-control/stats-date-control-picker.tsx (hunk 1 above)
Comment (kangzj) [reply client/components/stats-date-control/stats-date-control-picker.tsx:83]: I see https://github.com/Automattic/wp-calypso/pull/83336/commits/09902bf7d05a20da3e7ca1efce5e767452c647b6

File: client/components/stats-date-control/stats-date-control-picker-shortcuts.tsx [hunk 2]
```diff
@@ -1,40 +1,40 @@
...
 			<ul className="date-control-picker-shortcuts__list">
-				{ shortcutList.map( ( shortcut, idx ) => (
-					<li className={ classNameForShortcut( shortcut ) } key={ shortcut.id || idx }>
-						<Button
+				{ shortcutList.map( ( shortcut, idx ) => {
+					const isSelected = shortcut.id === currentShortcut;
+
+					return (
+						<li
+							className={ classNames( 'date-control-picker-shortcuts__shortcut', {
```
Comment (kangzj) [inline client/components/stats-date-control/stats-date-control-picker-shortcuts.tsx:20]: Hi @a8ck3n,

Just FYI I think it's simpler and easier to understand if written this way. `classNames` is so well known for Reactjs developers, and is kinda a standard.

I always appreciate readability over simplicity. If it's simple and readable then it's better than verbosity and readable. Verbosity means more code to write and read - IMHO in most cases, it doesn't have as good readability. 😄

Comment (a8ck3n) [reply client/components/stats-date-control/stats-date-control-picker-shortcuts.tsx:20]: Agreed on the use of `classNames` here, although I don't feel strongly about it.

I disagree when it comes to inlining logic in the JSX though. As someone newer to JSX, I definitely find it harder to parse when mixing logic and presentation like this. I'd say the same for anonymous functions in general. When they move beyond a single line or two, they feel like an abuse of the pattern. I'd like to be able to update logic without accidentally altering layout which is easier to reason about when having clearer separation.

I also find it harder to parse DIFFs when we inline our logic. Partially this is due to the auto-formatting we apply but it means that as things indent, they are more likely to be reformatted and pollute the resulting DIFF.

**Pull request merged**